test/rules/no-mocha-arrows.js

```diff
@@ -23,28 +23,39 @@ ruleTester.run('no-mocha-arrows', rules['no-mocha-arrows'], {
         {
             code: 'it(() => { assert(something, false); })',
             parserOptions: { ecmaVersion: 6 },
-            errors: [ { message: expectedErrorMessage, column: 1, line: 1 } ]
+            errors: [ { message: expectedErrorMessage, column: 1, line: 1 } ],
+            output: 'it(function () { assert(something, false); })'
         },
         {
             code: 'it(() => { assert(something, false); })',
             globals: [ 'it' ],
             parserOptions: { ecmaVersion: 6 },
-            errors: [ { message: expectedErrorMessage, column: 1, line: 1 } ]
+            errors: [ { message: expectedErrorMessage, column: 1, line: 1 } ],
+            output: 'it(function () { assert(something, false); })'
         },
         {
             code: 'it(() => assert(something, false))',
             parserOptions: { ecmaVersion: 6 },
-            errors: [ { message: expectedErrorMessage, column: 1, line: 1 } ]
+            errors: [ { message: expectedErrorMessage, column: 1, line: 1 } ],
+            output: 'it(function () { assert(something, false); })'
```

Comment:
The output should be
`it(function () { return assert(something, false); })`

Imagine the user wrote a test for function that returns a Promise, and they just want to check that it resolves.

```js
it(() => lib.resolves());
```

If it's changed to

```js
it(() => {
  lib.resolves();
});
```

then the test will passed even if the Promise rejects. It's much safer to change it to

```js
it(function() {
  return lib.resolves();
});
```